core/src/main.rs

```diff
@@ -135,6 +139,7 @@ async fn main() -> Result<(), CoordinatorError> {
 
     // Register metrics samplers
     RaftMetricsSampler::new(global_state.clone()).register(&system_clock).await?;
+    OnboardingMetricsSampler::new(global_state.clone()).register(&system_clock).await?;
```

Comment:
maybe break these out into a helper to colocate for future? maybe `setup_metrics_samplers`